public/js/clients/chrome.js

```diff
@@ -47,140 +98,90 @@ function presentTabs(tabs) {
-  },
-
-  scriptFailedToParse: function() {
-    // needed for debugging nytimes
-    console.log("SCRIPT FAILED TO PARSED", arguments);
-  },
+    scriptFailedToParse() {
+      // needed for debugging nytimes
+      console.log("SCRIPT FAILED TO PARSED", arguments);
+    },
 
-  paused: function(callFrames, reason, data, hitBreakpoints, asyncStackTrace) {
-    const frames = callFrames.map(frame => {
-      return Frame({
-        id: frame.callFrameId,
-        displayName: frame.functionName,
-        location: Location({
-          sourceId: frame.functionLocation.scriptId,
-          line: frame.functionLocation.lineNumber + 1,
-          column: frame.functionLocation.columnNumber
-        })
+    paused(callFrames, reason, data, hitBreakpoints, asyncStackTrace) {
+      const frames = callFrames.map(frame => {
+        return Frame({
+          id: frame.callFrameId,
+          displayName: frame.functionName,
+          location: Location({
+            sourceId: frame.functionLocation.scriptId,
+            line: frame.functionLocation.lineNumber + 1,
+            column: frame.functionLocation.columnNumber
+          })
+        });
       });
-    });
 
-    const frame = frames[0];
-    const why = Object.assign({}, {
-      type: reason
-    }, data);
+      const frame = frames[0];
+      const why = Object.assign({}, {
+        type: reason
+      }, data);
 
-    actions.paused({ frame, why });
-    actions.loadedFrames(frames);
-  },
+      actions.paused({ frame, why });
+      actions.loadedFrames(frames);
+    },
 
-  resumed: function() {
-    actions.resumed();
-  }
-});
+    resumed: function() {
+      actions.resumed();
+    }
+  };
+}
 
-function onConnection(connection, actions) {
-  const ws = connection._socket;
-  connectionAgents = connection._agents;
+function initPage(actions) {
+  const conn = currentConnection;
+  const ws = conn._socket;
+  currentDebuggerAgent = conn._agents.Debugger;
+  currentRuntimeAgent = conn._agents.Runtime;
   ws.onopen = function() {
   };
   ws.onmessage = function(e) {
-    connection._onMessage(e);
+    conn._onMessage(e);
   };
 
-  connection.registerDispatcher("Debugger", debuggerDispatcher(actions));
-  const debuggerAgent = getAgent("Debugger");
-  debuggerAgent.enable();
-  debuggerAgent.setPauseOnExceptions("none");
-  debuggerAgent.setAsyncCallStackDepth(0);
+  conn.registerDispatcher("Debugger", makeDispatcher(actions));
 
-  const runtimeAgent = getAgent("Runtime");
-  runtimeAgent.enable();
-  runtimeAgent.run();
-}
+  currentDebuggerAgent.enable();
+  currentDebuggerAgent.setPauseOnExceptions("none");
+  currentDebuggerAgent.setAsyncCallStackDepth(0);
 
-function debugChromeTab(tab, actions) {
-  bootstrap(InspectorBackend);
-  WebSocketConnection.Create(
-    tab.webSocketDebuggerUrl,
-    conn => onConnection(conn, actions)
-  );
-  return Promise.resolve();
+  currentRuntimeAgent.enable();
+  currentRuntimeAgent.run();
```

Comment:
The *current* naming convention in the clients seems unnecessary. I understand that we might have a different client connection at some point, but if we're not keeping past agents, threads, etc, current seems redundant. (small nit)